driver/normalizer/normalizer.go

```diff
@@ -793,25 +866,128 @@ var Normalizers = []Mapping{
 		},
 	)),
 
-	funcDefMap("MethodDeclaration", true, Obj{
-		// FIXME(dennwc): driver drops them currently
-		"Arity":                      Any(),
-		"ConstraintClauses":          Any(),
-		"ExplicitInterfaceSpecifier": Any(),
-		"TypeParameterList":          Any(),
-	}),
-	funcDefMap("ConstructorDeclaration", false, Obj{
-		// FIXME(dennwc): driver drops them currently
-		"Initializer": Any(),
-	}),
+	funcDefMap("MethodDeclaration", true,
+		Obj{
+			// number of parameters - safe to ignore
+			"Arity":                      Any(),
+			"ExplicitInterfaceSpecifier": Is(nil),
+			"ConstraintClauses": Cases("caseConstraint",
+				Arr(),
+				NotEmpty(Var("constraints")),
+			),
+			"TypeParameterList": Cases("caseTypeParams",
+				Is(nil),
+				Arr(),
+				NotEmpty(Var("typeParams")),
+			),
+		},
+		Cases("caseTypeParams",
+			Is(nil),
+			Is(nil),
+			NotEmpty(Var("typeParams")),
+		),
+		Cases("caseConstraint",
+			Is(nil),
+			NotEmpty(Var("constraints")),
+		),
+	),
+	// ConstructorDeclaration is similar to MethodDeclaration, but it may include a
+	// base class initializer that require a special transformation.
+	MapSemantic("ConstructorDeclaration", uast.FunctionGroup{}, MapObj(
+		Obj{
+			// Same as to MethodDeclaration above
+			"Identifier": Var("name"),
+			"ParameterList": Obj{
+				uast.KeyType:         String("ParameterList"),
+				uast.KeyPos:          Any(),
+				"OpenParenToken":     Any(),
```

Comment:
Can this really be Any? Isn't it a closed set?

Reply:
`Any` tells that any node will pass the validation, but it stores no state, so this node will be dropped. And this is fine because we don't store parentheses positions in Semantic. We technically can, but this is work for a later time.

Reply:
Makes sense, thanks for the explanation.